KarhooUISDK/Screens/CheckoutScreen/Components/AddPaymentView/KarhooAddPaymentPresenter.swift

```diff
@@ -59,13 +62,40 @@ final class KarhooAddPaymentPresenter: AddPaymentPresenter {
     private func handleAddCardFlow(result: CardFlowResult) {
         switch result {
         case .didAddPaymentMethod(let nonce):
+            reportCardAuthorisationSuccess()
             view.set(nonce: nonce)
         case .didFailWithError(let error):
+            reportCardAuthorisationFailure(message: error?.message ?? "")
             (view.parentViewController as? BaseViewController)?.showAlert(title: UITexts.Errors.somethingWentWrong,
                                                                           message: error?.message ?? "", error: error)
         default: break
         }
     }
+    
+    internal func setQuote(_ quote: Quote){
+        self.quote = quote
+    }
+
+    // Analytics
```

Comment:
If you wanted to make it as mark, then please use MARK formatting. If it should be simple comment regarding the method, it's not very helpful in current shape :)